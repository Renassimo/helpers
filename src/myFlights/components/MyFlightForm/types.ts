import { MyFlightAttributes } from '@/myFlights/types';
import { Matcher } from '@/common/types/matchers';

export default interface MyFlightFormFieldProps {
  name: keyof MyFlightAttributes;
  label: string;
  options?: string[];
  matchers?: Matcher;
  type?: string;
  disabled?: boolean;
  isDate?: boolean;
}
