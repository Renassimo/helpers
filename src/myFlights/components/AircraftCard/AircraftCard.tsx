import { showTimePassed, showWhen } from '@/common/utils/dayjs';

import { Avia } from '@/avia/types/avia';

import AviaCard from '@/avia/components/AviaCard';

const AircraftCard = ({
  data,
  onClick,
  chosen = false,
}: {
  data: Avia.AircraftData;
  onClick: () => void;
  chosen?: boolean;
}) => {
  const { attributes } = data;
  const {
    registration,
    serial,
    airlineName,
    typeName,
    productionLine,
    modelCode,
    model,
    photoUrl,
    firstFlightDate,
    rolloutDate,
    deliveryDate,
  } = attributes;

  const date = firstFlightDate ?? rolloutDate ?? deliveryDate;

  const additionalContent: (string | null)[] = [
    airlineName,
    typeName || productionLine,
    modelCode || model,
    serial && `CN: ${serial}`,
    date && `From ${showWhen(date, false)} (${showTimePassed(date)})`,
  ];

  return (
    <AviaCard
      onClick={onClick}
      chosen={chosen}
      imageUrl={photoUrl ?? '/images/plane.png'}
      imageAlt={registration}
      title={registration}
      additionalContent={additionalContent}
    />
  );
};

export default AircraftCard;
