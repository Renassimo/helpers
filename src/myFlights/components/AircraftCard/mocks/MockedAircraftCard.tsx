const MockedAircraftCard = jest.fn(({ chosen }) => (
  <div>MockedAircraftCard - {chosen && 'chosen'}</div>
));

export default MockedAircraftCard;
