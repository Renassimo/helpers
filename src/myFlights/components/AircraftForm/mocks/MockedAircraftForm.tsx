const MockedAircraftForm = jest.fn(({ registration }) => (
  <div>MockedAircraftForm - {registration}</div>
));

export default MockedAircraftForm;
