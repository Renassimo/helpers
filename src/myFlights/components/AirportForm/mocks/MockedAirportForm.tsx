const MockedAirportForm = jest.fn(({ title }) => (
  <div>MockedAirportForm - {title}</div>
));

export default MockedAirportForm;
