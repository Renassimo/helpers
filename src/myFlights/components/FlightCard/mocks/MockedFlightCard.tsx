const MockedFlightCard = jest.fn(({ data }) => (
  <div>MockedFlightCard - {data?.id}</div>
));

export default MockedFlightCard;
