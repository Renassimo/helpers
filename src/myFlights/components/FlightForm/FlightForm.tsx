import { KeyboardEvent } from 'react';

import Box from '@mui/material/Box';
import IconButton from '@mui/material/IconButton';
import SearchIcon from '@mui/icons-material/Search';

import useMyFlightsContext from '@/myFlights/contexts/hooks/useMyFlightsContext';

import useInputValue from '@/common/hooks/useInputValue';

import FlightCard from '@/myFlights/components/FlightCard';
import AviaInput from '@/avia/components/AviaInput';
import DatePicker from '@/common/components/DateInput';

const FlightForm = () => {
  const {
    flightsResult: {
      flights,
      chosenFlight,
      retreiveFlights,
      chooseFlight,
      clearChosenFlight,
      loading,
    },
  } = useMyFlightsContext();

  const [flightValue, setFlightValue] = useInputValue<string>('');
  const [flightDate, setFlightDate] = useInputValue<string | null>(null);

  const onSubmit = async () => {
    await retreiveFlights(flightValue, flightDate);
  };

  const lengthOk = flightValue.length > 2;

  const onEnter = (event: KeyboardEvent) => {
    if (event.key === 'Enter' && lengthOk) onSubmit();
  };

  return (
    <Box>
      {!chosenFlight && (
        <>
          <AviaInput
            title="Flight number"
            value={flightValue}
            setValue={setFlightValue}
            disabled={loading}
            onKeyDown={onEnter}
            adornments={
              <IconButton
                size="small"
                onClick={onSubmit}
                disabled={loading || !lengthOk}
                aria-label="Search"
              >
                <SearchIcon />
              </IconButton>
            }
          />
          <Box mt={2}>
            <DatePicker
              value={flightDate}
              setValue={(newValue) => setFlightDate(newValue)}
              placeholder="Flight date"
              disabled={loading}
              fullWidth
            />
          </Box>
        </>
      )}
      {flights && !chosenFlight && (
        <Box display="flex" overflow="auto">
          {flights.map((flight, index) => (
            <Box
              key={flight.id + index}
              mr={1}
              ml={index === 0 ? 1 : undefined}
            >
              <FlightCard
                data={flight}
                onClick={() => chooseFlight(flight.id)}
              />
            </Box>
          ))}
        </Box>
      )}
      {chosenFlight && (
        <FlightCard data={chosenFlight} onClick={clearChosenFlight} chosen />
      )}
    </Box>
  );
};

export default FlightForm;
