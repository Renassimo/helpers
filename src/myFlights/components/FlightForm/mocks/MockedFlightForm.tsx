const MockedFlightForm = jest.fn(() => <div>MockedFlightForm</div>);

export default MockedFlightForm;
