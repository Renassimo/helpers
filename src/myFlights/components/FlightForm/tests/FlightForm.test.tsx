import userEvent from '@testing-library/user-event';
import renderWithTheme from '@/common/tests/helpers/renderWithTheme';

import { UseFlightsResult } from '@/myFlights/types';

import useInputValue from '@/common/hooks/useInputValue';

import useMyFlightsContext from '@/myFlights/contexts/hooks/useMyFlightsContext';

import AviaInput from '@/avia/components/AviaInput';
import AircraftCard from '@/myFlights/components/AircraftCard';

import MockedAviaInput from '@/avia/components/AviaInput/mocks';
import MockedAircraftCard from '@/myFlights/components/AircraftCard/mocks';

import { mockedDeserializedFlights } from '@/avia/types/avia/mocks';

import FlightForm from '../FlightForm';

jest.mock('@/common/hooks/useInputValue');
jest.mock('@/avia/components/AviaInput');
jest.mock('@/myFlights/contexts/hooks/useMyFlightsContext');
jest.mock('@/myFlights/components/AircraftCard');

describe('FlightForm', () => {
  const mockedSetValue = jest.fn();
  const mockUseInputValue = (value = '') =>
    jest.fn(() => [value, mockedSetValue]);

  const retreiveFlights = jest.fn();
  const chooseFlight = jest.fn();
  const clearChosenFlight = jest.fn();

  const flightsResult = {
    flights: [mockedDeserializedFlights[0], mockedDeserializedFlights[0]],
    chosenFlight: mockedDeserializedFlights[0],
    retreiveFlights,
    chooseFlight,
    clearChosenFlight,
    loading: false,
  } as unknown as UseFlightsResult;

  const mockUseMyFlightsContext = (props: Partial<UseFlightsResult> = {}) =>
    jest.fn(() => ({ flightsResult: { ...flightsResult, ...props } }));

  beforeEach(() => {
    (AviaInput as unknown as jest.Mock).mockImplementation(MockedAviaInput);
    (AircraftCard as unknown as jest.Mock).mockImplementation(
      MockedAircraftCard
    );
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('renders successfully', () => {
    // Arange
    (useInputValue as unknown as jest.Mock).mockImplementation(
      mockUseInputValue()
    );
    (useMyFlightsContext as unknown as jest.Mock).mockImplementation(
      mockUseMyFlightsContext()
    );
    // Act
    const { baseElement } = renderWithTheme(<FlightForm />);
    // Assert
    expect(baseElement).toMatchSnapshot();
  });

  describe('when no chosenFlight', () => {
    const contextProps = { chosenFlight: null };

    test('renders successfully', () => {
      // Arange
      (useInputValue as unknown as jest.Mock).mockImplementation(
        mockUseInputValue()
      );
      (useMyFlightsContext as unknown as jest.Mock).mockImplementation(
        mockUseMyFlightsContext(contextProps)
      );
      // Act
      const { baseElement } = renderWithTheme(<FlightForm />);
      // Assert
      expect(baseElement).toMatchSnapshot();
    });

    describe('and when value is not empty', () => {
      const value = 'Val';

      test('renders successfully', async () => {
        // Arange
        (useInputValue as unknown as jest.Mock).mockImplementation(
          mockUseInputValue(value)
        );
        (useMyFlightsContext as unknown as jest.Mock).mockImplementation(
          mockUseMyFlightsContext(contextProps)
        );
        // Act
        const { baseElement } = renderWithTheme(<FlightForm />);
        // Assert
        expect(baseElement).toMatchSnapshot();
      });

      describe('and clicks to search', () => {
        test('calls retreiveFlights', async () => {
          // Arange
          (useInputValue as unknown as jest.Mock).mockImplementation(
            mockUseInputValue(value)
          );
          (useMyFlightsContext as unknown as jest.Mock).mockImplementation(
            mockUseMyFlightsContext(contextProps)
          );
          const { getByLabelText } = renderWithTheme(<FlightForm />);
          // Act
          await userEvent.click(getByLabelText('Search'));
          // Assert
          expect(retreiveFlights).toBeCalledWith('Val');
        });
      });
    });
  });

  describe('when no aircrafts', () => {
    test('renders successfully', () => {
      // Arange
      (useInputValue as unknown as jest.Mock).mockImplementation(
        mockUseInputValue()
      );
      (useMyFlightsContext as unknown as jest.Mock).mockImplementation(
        mockUseMyFlightsContext({ chosenFlight: null, flights: null })
      );
      // Act
      const { baseElement } = renderWithTheme(<FlightForm />);
      // Assert
      expect(baseElement).toMatchSnapshot();
    });
  });
});
