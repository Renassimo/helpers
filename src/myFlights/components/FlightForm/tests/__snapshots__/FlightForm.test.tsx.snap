// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightForm renders successfully 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-98fxvp-MuiPaper-root-MuiCard-root"
      >
        <button
          aria-label="Close"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1nxn43-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="Action area"
          class="MuiButtonBase-root Mui-disabled MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <img
            alt="FZ 1839"
            class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
            height="120"
            src="https://farm66.staticflickr.com/65535/51788228697_3c96482599_z.jpg"
          />
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-ejgmvm-MuiTypography-root"
            >
              FZ 1839
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
            >
              Dubai - Warsaw Chopin
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
            >
              DXB/OMDB - WAW/EPWA
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
            >
              flydubai
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
            >
              Boeing 737 MAX 8
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
            >
              A6-FMB
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
            >
              4157 km
            </p>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
          />
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`FlightForm when no aircrafts renders successfully 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <div>
        MockedAviaInput - 
        Flight number
         - 
         - 
         - 
         -
         
         - 
        <button
          aria-label="Search"
          class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`FlightForm when no chosenFlight and when value is not empty renders successfully 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <div>
        MockedAviaInput - 
        Flight number
         - 
         - 
         - 
        Val
         -
         
         - 
        <button
          aria-label="Search"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiBox-root css-ohyvb6"
      >
        <div
          class="MuiBox-root css-hnroeo"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-98fxvp-MuiPaper-root-MuiCard-root"
          >
            <button
              aria-label="Action area"
              class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
              tabindex="0"
              type="button"
            >
              <img
                alt="FZ 1839"
                class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
                height="120"
                src="https://farm66.staticflickr.com/65535/51788228697_3c96482599_z.jpg"
              />
              <div
                class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-ejgmvm-MuiTypography-root"
                >
                  FZ 1839
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                >
                  Dubai - Warsaw Chopin
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                >
                  DXB/OMDB - WAW/EPWA
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                >
                  flydubai
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                >
                  Boeing 737 MAX 8
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                >
                  A6-FMB
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                >
                  4157 km
                </p>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiBox-root css-12z0wuy"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-98fxvp-MuiPaper-root-MuiCard-root"
          >
            <button
              aria-label="Action area"
              class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
              tabindex="0"
              type="button"
            >
              <img
                alt="FZ 1839"
                class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
                height="120"
                src="https://farm66.staticflickr.com/65535/51788228697_3c96482599_z.jpg"
              />
              <div
                class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-ejgmvm-MuiTypography-root"
                >
                  FZ 1839
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                >
                  Dubai - Warsaw Chopin
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                >
                  DXB/OMDB - WAW/EPWA
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                >
                  flydubai
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                >
                  Boeing 737 MAX 8
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                >
                  A6-FMB
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                >
                  4157 km
                </p>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`FlightForm when no chosenFlight renders successfully 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <div>
        MockedAviaInput - 
        Flight number
         - 
         - 
         - 
         -
         
         - 
        <button
          aria-label="Search"
          class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </button>
      </div>
      <div
        class="MuiBox-root css-ohyvb6"
      >
        <div
          class="MuiBox-root css-hnroeo"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-98fxvp-MuiPaper-root-MuiCard-root"
          >
            <button
              aria-label="Action area"
              class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
              tabindex="0"
              type="button"
            >
              <img
                alt="FZ 1839"
                class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
                height="120"
                src="https://farm66.staticflickr.com/65535/51788228697_3c96482599_z.jpg"
              />
              <div
                class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-ejgmvm-MuiTypography-root"
                >
                  FZ 1839
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                >
                  Dubai - Warsaw Chopin
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                >
                  DXB/OMDB - WAW/EPWA
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                >
                  flydubai
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                >
                  Boeing 737 MAX 8
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                >
                  A6-FMB
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                >
                  4157 km
                </p>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiBox-root css-12z0wuy"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-98fxvp-MuiPaper-root-MuiCard-root"
          >
            <button
              aria-label="Action area"
              class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
              tabindex="0"
              type="button"
            >
              <img
                alt="FZ 1839"
                class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
                height="120"
                src="https://farm66.staticflickr.com/65535/51788228697_3c96482599_z.jpg"
              />
              <div
                class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-ejgmvm-MuiTypography-root"
                >
                  FZ 1839
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                >
                  Dubai - Warsaw Chopin
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                >
                  DXB/OMDB - WAW/EPWA
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                >
                  flydubai
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                >
                  Boeing 737 MAX 8
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                >
                  A6-FMB
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                >
                  4157 km
                </p>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
